src/Console/ConfigurationResolver.php

```diff
@@ -390,13 +392,51 @@ private function resolveConfig()
      */
     private function resolveConfigPath()
     {
-        if (is_file($this->path)) {
-            $this->config->finder(new \ArrayIterator(array(new \SplFileInfo($this->path))));
-        } elseif ($this->isStdIn) {
+        if ($this->isStdIn) {
             $this->config->finder(new \ArrayIterator(array(new StdinFileInfo())));
-        } elseif (null !== $this->path) {
-            $this->config->setDir($this->path);
+
+            return;
+        }
+
+        if (null === $this->path) {
+            return;
         }
+
+        // - when we already have a valid finder - create intersection iterator of current finder and provided path
+        // - when we don't have valid finder - prepare new iterator
+        $iterator = null;
+        $currentFinder = $this->config->getFinder();
+        $path = realpath($this->path);
+        $isFile = is_file($path);
+
+        if ($isFile) {
+            $callback = function (\SplFileInfo $current) use ($path) {
+                return $current->getRealPath() === $path;
+            };
+        } else {
+            $path .= DIRECTORY_SEPARATOR;
+            $callback = function (\SplFileInfo $current) use ($path) {
+                return 0 === strpos($current->getRealPath(), $path);
+            };
+        }
+
+        try {
+            $iterator = new \CallbackFilterIterator(
+                $currentFinder instanceof \IteratorAggregate ? $currentFinder->getIterator() : $currentFinder,
+                $callback
+            );
+        } catch (\LogicException $e) {
+            $iterator = $isFile
+                ? new \ArrayIterator(array(new SymfonySplFileInfo(
+                    $this->path,
+                    substr(dirname($this->path), strlen($this->cwd) + 1),
+                    substr($this->path, strlen($this->cwd) + 1)
+                )))
+                : $currentFinder->in($this->path)
```

Comment:
Isn't it possible that `$currentFinder` is not a symfony finder instance, so the method `in` would not exist?